tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -29,459 +29,467 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
-				return strings.Contains(output, operator)
-			})
-		}
-	}
-
-	cleanPreSetup := func() {
-		helper.DeleteProject(commonVar.Project)
-	}
-
-	Context("When Operators are installed in the cluster", func() {
+	Context("Operators are installed in the cluster", func() {
 
+		var operators string
 		JustBeforeEach(func() {
-			preSetup()
+			// wait till oc can see the all operators installed by setup script in the namespace
+			odoArgs := []string{"catalog", "list", "services"}
+			operators := []string{"etcdoperator", "service-binding-operator"}
+			for _, operator := range operators {
+				helper.WaitForCmdOut("odo", odoArgs, 5, true, func(output string) bool {
+					return strings.Contains(output, operator)
+				})
+			}
 		})
 
 		JustAfterEach(func() {
-			cleanPreSetup()
-		})
-
-		It("should list operators installed in the namespace", func() {
-			stdOut := helper.CmdShouldPass("odo", "catalog", "list", "services")
-			helper.MatchAllInOutput(stdOut, []string{"Services available through Operators", "etcdoperator"})
-		})
-
-		It("should describe an installed operator with json output", func() {
-			operators := helper.CmdShouldPass("odo", "catalog", "list", "services")
-			etcdOperator := regexp.MustCompile(`etcdoperator\.*[a-z][0-9]\.[0-9]\.[0-9]-clusterwide`).FindString(operators)
-			etcdCluster := fmt.Sprintf("%s/EtcdCluster", etcdOperator)
-
-			output := helper.CmdShouldPass("odo", "catalog", "describe", "service", etcdCluster)
-			Expect(output).To(ContainSubstring("Kind: EtcdCluster"))
-			outputJSON := helper.CmdShouldPass("odo", "catalog", "describe", "service", etcdCluster, "-o", "json")
-			values := gjson.GetMany(outputJSON, "spec.kind", "spec.displayName")
-			expected := []string{"EtcdCluster", "etcd Cluster"}
-			Expect(helper.GjsonMatcher(values, expected)).To(Equal(true))
+			helper.DeleteProject(commonVar.Project)
 		})
 
-		It("should not allow creating service without valid context, and fail for interactive mode", func() {
+		It("should not allow creating service without valid context", func() {
 			stdOut := helper.CmdShouldFail("odo", "service", "create")
 			Expect(stdOut).To(ContainSubstring("service can be created/deleted from a valid component directory only"))
-
-			helper.CmdShouldPass("odo", "create", "nodejs")
-			stdOut = helper.CmdShouldFail("odo", "service", "create")
-			Expect(stdOut).To(ContainSubstring("odo doesn't support interactive mode for creating Operator backed service"))
 		})
-	})
-
-	Context("When creating and deleting an operator backed service", func() {
 
-		JustBeforeEach(func() {
-			preSetup()
+		It("should list operators installed in the namespace", func() {
+			operators = helper.CmdShouldPass("odo", "catalog", "list", "services")
+			helper.MatchAllInOutput(operators, []string{"Services available through Operators", "etcdoperator"})
 		})
 
-		JustAfterEach(func() {
-			cleanPreSetup()
-		})
+		Context("the etcd operator is installed", func() {
```

Comment:
Is it necessary to specify the operator name? Can this description not be generic?
I ask this because I heard that we are trying the reduce the usage/drop the support for etcd operator. Plus, these tests are meant to test the operator and not this specific one.
I am not suggesting we change test code, just the description.